tests/scripts/configs_e2e/configs_e2e.go

```diff
@@ -0,0 +1,408 @@
+	dir := path.Dir(dst)
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+		return fmt.Errorf("failed to create directory %q when copying %q to %q: %w", dir, src, dst, err)
+	}
+	i, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open %q for reading: %w", src, err)
+	}
+	defer i.Close()
+
+	o, err := os.Create(dst)
+	if err != nil {
+		return fmt.Errorf("failed to open %q for writing: %w", dst, err)
+	}
+	defer o.Close()
+
+	if _, err := io.Copy(o, i); err != nil {
+		return fmt.Errorf("error while copying the contents of %q to %q: %w", src, dst, err)
+	}
+	return nil
+}
+
+func createWorkspaceFile(m *rbeconfigsgen.Manifest, configTarballURL string, outputDir string) error {
+	o, err := os.Create(path.Join(outputDir, "WORKSPACE"))
+	if err != nil {
+		return fmt.Errorf("unable to create WORKSPACE file in %q: %w", outputDir, err)
+	}
+	defer o.Close()
+	data := struct {
+		ConfigsTarballURL    string
+		ConfigsTarballDigest string
+	}{
+		ConfigsTarballURL:    configTarballURL,
+		ConfigsTarballDigest: m.ConfigsTarballDigest,
+	}
+	if err := workspaceTemplate.Execute(o, &data); err != nil {
+		return fmt.Errorf("error writing Bazel WORKSPACE file in %q: %w", outputDir, err)
+	}
+	log.Printf("Generated WORKSPACE file in %q.", outputDir)
+	return nil
+}
+
+func createBazelrcFile(m *rbeconfigsgen.Manifest, configTarballURL, outputDir, rbeInst string) error {
+	o, err := os.Create(path.Join(outputDir, ".bazelrc"))
+	if err != nil {
+		return fmt.Errorf("unable to open .bazelrc file for writing in %q: %w", outputDir, err)
+	}
+	defer o.Close()
+
+	fmt.Fprintf(o, `
+# .bazelrc generated for:
+#   Bazel %s
+#   Toolchain Container %s (sha256:%s)
+#   Configs Tarball URL %s (sha256:%s)
+`, m.BazelVersion, m.ToolchainContainer, m.ImageDigest, configTarballURL, m.ConfigsTarballDigest)
+	fmt.Fprintf(o, `
+build:remote --jobs=6
+build:remote --remote_instance_name=%s
+
+# C++ toolchain & default platform configuration.
+build:remote --crosstool_top=@rbe_default//cc:toolchain
+build:remote --action_env=BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1
+build:remote --extra_toolchains=@rbe_default//config:cc-toolchain
+build:remote --extra_execution_platforms=@rbe_default//config:platform
+build:remote --host_platform=@rbe_default//config:platform
+build:remote --platforms=@rbe_default//config:platform
+
+build:remote --define=EXECUTOR=remote
+build:remote --remote_executor=grpcs://remotebuildexecution.googleapis.com
+
+# Enforce stricter environment rules, which eliminates some non-hermetic
+# behavior and therefore improves both the remote cache hit rate and the
+# correctness and repeatability of the build.
+build:remote --incompatible_strict_action_env=true
+
+build:remote --remote_timeout=3600
+
+# Enable authentication. This will pick up application default credentials by
+# default. You can use --google_credentials=some_file.json to use a service
+# account credential instead.
+build:remote --google_default_credentials=true
+`, rbeInst)
```

Comment:
Consider moving this line up to :188 and starting a third Fprintf.

Reply:
Done